### PR TITLE
PIP-1549-s3-md5

### PR DESCRIPTION
**Pull request by paul-sud**

### CODE REVIEW

File: functions/initial_md5_hash/app.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    object_resource_name = ARN(
+        [i for i in resources if i["type"] == "AWS::S3::Object"][0]["ARN"]
+    ).name
+    bucket, *key_parts = object_resource_name.split("/")
+    key = "/".join(key_parts)
+    key = unquote_plus(key, encoding="utf-8")
+    object = _get_object(bucket=bucket, key=key)
+    etag = object["ETag"].strip('"')
+    # Fast path, ETag is the md5sum
+    if len(etag) == 32 and "-" not in etag:
```
Comment (ottojolanki) [inline functions/initial_md5_hash/app.py]: Nitty nit: md5sum is length 32 and alphanumeric, so you should check if it matches to something like `r"^[a-z0-9]{32}"`

**Review by ottojolanki — CHANGES_REQUESTED** [review]
Looks very good, maybe be more specific in testing if the tag content is an md5 sum is the only thing.

File: functions/initial_md5_hash/app.py (hunk 1 above)
Comment (paul-sud) [reply functions/initial_md5_hash/app.py]: Yeah changes this to check `re.match(r"^[a-f0-9]{32}$", etag) is not None`

**Review by ottojolanki — APPROVED** [review]
LGTM

**Pull request merged**